File: app/src/main/java/com/github/rooneyandshadows/lightbulb/easyrecyclerviewdemo/demo/adapters/LabelsAdapter.java
package com.github.rooneyandshadows.lightbulb.easyrecyclerviewdemo.demo.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.github.rooneyandshadows.lightbulb.commons.utils.ResourceUtils;
import com.github.rooneyandshadows.lightbulb.easyrecyclerviewdemo.R;
import com.github.rooneyandshadows.lightbulb.easyrecyclerviewdemo.demo.models.DemoModel;
import com.github.rooneyandshadows.lightbulb.recycleradapters.EasyAdapterConfiguration;
import com.github.rooneyandshadows.lightbulb.recycleradapters.EasyRecyclerAdapter;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class LabelsAdapter extends EasyRecyclerAdapter<DemoModel> {

    public LabelsAdapter() {
        super(new EasyAdapterConfiguration<>());
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new LabelItemViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.demo_list_item_label_layout, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder viewHolder, int position) {
        LabelItemViewHolder holder = (LabelItemViewHolder) viewHolder;
        holder.bindData(getItem(position));
    }

    @Override
    public void onViewRecycled(@NonNull RecyclerView.ViewHolder viewHolder) {
        super.onViewRecycled(viewHolder);
        LabelItemViewHolder holder = (LabelItemViewHolder) viewHolder;
        holder.recycle();
    }

    public static class LabelItemViewHolder extends RecyclerView.ViewHolder {
        private final RelativeLayout container;
        private final TextView textView;

        LabelItemViewHolder(View view) {
            super(view);
            container = (RelativeLayout) view;
            textView = view.findViewById(R.id.labelTextView);
        }

        private void bindData(DemoModel model) {
            container.setBackground(ResourceUtils.getDrawable(container.getContext(), R.drawable.bg_label_item));
            textView.setText(model.getTitle());
        }

        private void recycle() {
            container.setBackground(null);
        }
    }
}

File: app/src/main/java/com/github/rooneyandshadows/lightbulb/easyrecyclerviewdemo/demo/views/adapters/LabelsAdapter.kt
package com.github.rooneyandshadows.lightbulb.easyrecyclerviewdemo.demo.views.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.RelativeLayout
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.github.rooneyandshadows.lightbulb.commons.utils.ResourceUtils
import com.github.rooneyandshadows.lightbulb.easyrecyclerviewdemo.R
import com.github.rooneyandshadows.lightbulb.easyrecyclerviewdemo.demo.models.DemoModel
import com.github.rooneyandshadows.lightbulb.recycleradapters.abstraction.EasyRecyclerAdapter
import com.github.rooneyandshadows.lightbulb.recycleradapters.implementation.collection.ExtendedCollection
import com.github.rooneyandshadows.lightbulb.recycleradapters.implementation.collection.ExtendedCollection.SelectableModes.SELECT_SINGLE

class LabelsAdapter : EasyRecyclerAdapter<DemoModel>() {
    override val collection: ExtendedCollection<DemoModel>
        get() = super.collection as ExtendedCollection<DemoModel>

    override fun createCollection(): ExtendedCollection<DemoModel> {
        return ExtendedCollection(this, SELECT_SINGLE)
    }

    @Override
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        return LabelItemViewHolder(
            LayoutInflater.from(parent.context)
                .inflate(R.layout.demo_list_item_label_layout, parent, false)
        )
    }

    @Override
    override fun onBindViewHolder(viewHolder: RecyclerView.ViewHolder, position: Int) {
        val holder = viewHolder as LabelItemViewHolder
        holder.bindData(collection.getItem(position)!!)
    }

    @Override
    override fun onViewRecycled(viewHolder: RecyclerView.ViewHolder) {
        super.onViewRecycled(viewHolder)
        val holder = viewHolder as LabelItemViewHolder
        holder.recycle()
    }

    class LabelItemViewHolder internal constructor(view: View) : RecyclerView.ViewHolder(view) {
        private val container: RelativeLayout
        private val textView: TextView

        init {
            container = view as RelativeLayout
            textView = view.findViewById(R.id.labelTextView)
        }

        fun bindData(model: DemoModel) {
            container.background =
                ResourceUtils.getDrawable(container.context, R.drawable.bg_label_item)
            textView.text = model.title
        }

        fun recycle() {
            container.background = null
        }
    }
}